Added helper scripts and updated README with example usage

mheadd opened the pull request:
# Add Bash Helper Scripts

This PR addresses #72 highlighting the need for bash helper scripts for the Census MCP commands.

## What This PR Adds

### Main Helper Script
- `census-mcp.sh` - Unified interface to all Census MCP tools with subcommands
- Provides consistent API key handling and validation
- Includes comprehensive help documentation
- Enhanced to support: `--descriptive` and `--predicates` parameters

### Individual Tool Scripts
- `list-tools.sh` - List available MCP tools
- `list-prompts.sh` - List available MCP prompts
- `list-datasets.sh` - Fetch all available Census datasets
- `fetch-dataset-geography.sh` - Get geography levels for a dataset
- `fetch-aggregate-data.sh` - With full parameter support including `descriptive` and `predicates`
- `resolve-geography-fips.sh` - Resolve geography names to FIPS codes
- `get-population-data.sh` - Get population data for geographic areas

### Tool Schema Analysis Integration
- Analyzed actual MCP tool schemas to ensure documentation accuracy
- Updated README to emphasize `fetch-aggregate-data` as PRIMARY tool for statistical queries
- Added proper workflow guidance (list-datasets -> resolve-geography -> fetch-data)
- Enhanced examples with real tool schema information

## Key Features

### API Key Handling
- All scripts check for `CENSUS_API_KEY` environment variable
- Display clear error messages when API key is missing
- Support for both exported environment variables and inline usage

### JSON Output Support
- Provided `--json` flag for all scripts to output only JSON (no descriptive text), ideal for piping to `jq` and other JSON processing tools
- Works with both individual scripts and main wrapper script

### User-Friendly Interface
- Simplified argument passing (no complex JSON required)
- Built-in help for each script (`--help`, `-h`, or no arguments)
- Input validation with clear error messages
- Usage examples included in help text

### Docker Management
- Automatic Docker service startup when needed
- Consistent container interaction across all scripts
- Proper error handling for Docker-related issues

### Consistent Structure
- All scripts follow the same pattern and conventions
- Proper argument validation and help documentation
- Standard exit codes and error handling

## Files Modified/Created

### New Files
- `scripts/census-mcp.sh` - Main helper script
- `scripts/list-tools.sh`
- `scripts/list-prompts.sh`
- `scripts/list-datasets.sh`
- `scripts/fetch-dataset-geography.sh`
- `scripts/fetch-aggregate-data.sh`
- `scripts/resolve-geography-fips.sh`
- `scripts/get-population-data.sh`

### Modified Files
- `README.md` - Updated with helper script documentation

## Implementation Notes

- All scripts are bash-compatible and tested on macOS
- Scripts automatically detect project directory structure
- Comprehensive input validation prevents common errors
- Consistent error handling and exit codes throughout
- Scripts are self-documenting with built-in help

annav-gov commented:
@mheadd thank you so much taking a look at our issues and creating this PR to create the bash helper scripts! We really appreciate your help and contribution to the project! We'll take a closer look and follow up with any feedback shortly.

Thanks again for jumping in and helping out!

annav-gov commented:
@curt-mitch-gov, mind being a second pair of eyes on the bash scripts?

curt-mitch-gov commented on scripts/list-tools.sh:
```diff
@@ -0,0 +1,55 @@
+fi
+
+# Change to project directory
+cd "$PROJECT_DIR"
```
We should include an `exit` option here:
```suggestion
cd "$PROJECT_DIR" || exit
```

curt-mitch-gov commented on scripts/resolve-geography-fips.sh:
```diff
@@ -0,0 +1,98 @@
+#!/bin/bash
+
+# Resolve Geography FIPS Helper Script
+# This script wraps the resolve-geography-fips tool to search for geographic areas
+
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+PROJECT_DIR="$(dirname "$SCRIPT_DIR")"
+
```
It looks like we can remove this redundant code:
```suggestion
```

curt-mitch-gov commented on scripts/resolve-geography-fips.sh:
```diff
@@ -0,0 +1,98 @@
+fi
+
+# Change to project directory
+cd "$PROJECT_DIR"
```
We should add an `exit` option here:
```suggestion
cd "$PROJECT_DIR" || exit
```

mheadd commented:
@curt-mitch-gov @annav-gov. This is now ready for re-review. Happy to keep working on this if there are additional changes or enhancements required. Thanks!

## Structural Changes

* Moved all helper scripts from scripts to dev subdirectory (8 scripts total)
* Created comprehensive README.md in dev subdirectory with complete documentation for all development tools
* Updated top-level README.md to remove raw bash/Docker commands and focus on production MCP usage

## Documentation Improvements
* Centralized development documentation in README.md with table of contents
* Validated help text consistency - ensured all script --help output matches README examples exactly
* Fixed documentation errors including incorrect command names and JSON processing examples
* Added working examples for all scripts with proper error handling patterns

## Comprehensive Testing & Validation
* Manually tested all 8 scripts using README examples to verify functionality
* Validated JSON processing and corrected jq command syntax throughout documentation
* Confirmed database setup requirements and documented the critical seeding step
F* ixed shell script safety with proper cd || exit error handling

## Results
* Clear separation of concerns: Main README focuses on production MCP usage, dev tools isolated in dev
* Accurate documentation: All examples work as documented
* Reliable development workflow: Scripts provide robust CLI interface for Census MCP development and testing

annav-gov commented on README.md:
```diff
@@ -111,97 +106,6 @@ To run tests, navigate to the `mcp-db/` directoruy and run `npm run test:db`.
-### List Datasets
-The `list-datasets` tool is used for fetching a subset of metadata for all datasets that are available in the Census Bureau's API. \
-It requires no arguments.
-
```
Please keep these lines. It's important documentation for what's available in the server.

annav-gov commented on README.md:
```diff
@@ -111,97 +106,6 @@ To run tests, navigate to the `mcp-db/` directoruy and run `npm run test:db`.
-* UCGID (Optional) - A string that restricts geography by Uniform Census Geography Identifier (UCGID), e.g. `0400000US41`
-* Predicates (Optional) - Filter options for the dataset, e.g. `'for': 'state*'`
-* Descriptive (Optional) - Adds variable labels to API response (default: `false`), e.g. `true`
-
```
Please keep these lines. It's important documentation for what's available in the server.

annav-gov commented on scripts/dev/census-mcp.sh:
```diff
@@ -0,0 +1,119 @@
+    echo "  list-datasets           List available Census datasets"
+    echo "  fetch-dataset-geography <dataset> [year]"
+    echo "                          Fetch geography levels for a dataset"
+    echo "  fetch-aggregate-data <dataset> <year> <variables> [for] [in] [ucgid] [--descriptive] [--predicates key:value]"
```
Same comment as in `fetch-aggregate-data.sh`: predicates should be a positional argument, rather than an optional flag.

annav-gov commented on scripts/dev/list-prompts.sh:
```diff
@@ -0,0 +1,55 @@
+            echo "  -h, --help Show this help message"
+            echo ""
+            echo "Environment:"
+            echo "  CENSUS_API_KEY must be set"
```
`list-prompts` doesn't actually require the API key; that was a documentation error. Please remove references to it here.

annav-gov commented on scripts/dev/list-tools.sh:
```diff
@@ -0,0 +1,55 @@
+            echo "  -h, --help Show this help message"
+            echo ""
+            echo "Environment:"
+            echo "  CENSUS_API_KEY must be set"
```
`list-tools` doesn't actually require the API key; that was a documentation error. Please remove references to it here.

annav-gov commented on scripts/dev/README.md:
```diff
@@ -0,0 +1,337 @@
+
+**Note: These scripts are for development and testing purposes only.** They are not the intended primary interface for interacting with the MCP server. For production use, connect your AI assistant client to the MCP server as described in the main [README.md](../../README.md).
+
+The Model Context Protocol (MCP) is designed for client-server interactions through MCP clients (like Claude Desktop, IDEs with MCP support, etc.). These bash helper scripts are provided to facilitate development, debugging, and testing workflows.
```
Appreciate that you added this. It's great.

annav-gov commented on scripts/dev/README.md:
```diff
@@ -0,0 +1,337 @@
+# JSON output examples (pipe to jq for processing)
+./list-datasets.sh --json | jq '.result.content[0].text | fromjson | keys'
+./resolve-geography-fips.sh 'Philadelphia' --json | jq '.result.content[0].text'
+./get-population-data.sh 'California' --json | jq '.result.content[0].text'
```
These examples are duplicated later in this README. Please consolidate.

annav-gov commented on scripts/dev/census-mcp.sh:
```diff
@@ -0,0 +1,119 @@
+    echo "  resolve-geography-fips <geography> [summary_level]"
+    echo "                          Resolve geography name to FIPS codes"
+    echo "  get-population-data <geography>"
+    echo "                          Get population data for a geography"
```
get-population-data is a prompt; it doesn't actually return the population data; just the prompt for it.

curt-mitch-gov reviewed (approved):
LGTM, thank you for this contribution and for your patience through the delayed responses! 🙏
